apps/studio/src/schemas/user.ts

```diff
@@ -0,0 +1,87 @@
+import { RoleType } from "~prisma/generated/generatedEnums"
+import { z } from "zod"
+
+import { offsetPaginationSchema } from "./pagination"
+
+export const createInputSchema = z.object({
+  siteId: z.number().min(1),
+  users: z.array(
+    z.object({
+      email: z.string().email(),
+      role: z.nativeEnum(RoleType).optional().default(RoleType.Editor),
+    }),
+  ),
+})
+
+export const createOutputSchema = z.array(
+  z.object({
+    id: z.string(),
+    email: z.string().email(),
+    role: z.nativeEnum(RoleType),
+  }),
+)
+
+export const deleteInputSchema = z.object({
+  siteId: z.number().min(1),
+  userId: z.string(),
+})
+
+export const deleteOutputSchema = z.object({
+  id: z.string(),
+  email: z.string().email(),
+})
+
+export const getUserInputSchema = z.object({
+  siteId: z.number().min(1),
+  userId: z.string(),
+})
+
+export const getUserOutputSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  email: z.string().email(),
+  role: z.nativeEnum(RoleType),
+  lastLoginAt: z.date().nullable(),
+})
+
+export const listInputSchema = z.object({
+  siteId: z.number().min(1),
+  getIsomerAdmins: z.boolean().optional().default(false),
```

Comment:
[nit] after reviewing the downstream PRs, this feels like it should be an array of enum of `RoleType` + `IsomerAdmin` and the backend will return all users whose roles are in this array

not a big deal but using booleans like this is a code smell

Reply:
`getIsomerAdmins ` is for us to return only isomer admins, or non isomer admins, for this tab here

<img width="868" alt="image" src="https://github.com/user-attachments/assets/2a308585-708f-4681-a98d-6fe59010fa52" />

i was contemplating:
1. whats the best params for this (couldn't think of one) and
2. whether i should spin this off as a standalone endpoint

but given that this is confusing, i think i will go with number 2!

oh and no need to return by `RoleType` for now since we are returning all users without any filtering logic